File: codewars/elixir/bit_counting.exs
# https://www.codewars.com/kata/526571aae218b8ee490006f4/train/elixir

defmodule CountBits do
  def bits(n) do
    n
      |> Integer.to_string(2)
      |> String.graphemes()
      |> Enum.filter(fn(b) ->  b == "1" end)
      |> Enum.count()
  end
end

IO.puts(CountBits.bits(0))
IO.puts(CountBits.bits(4))
IO.puts(CountBits.bits(7))
IO.puts(CountBits.bits(9))
IO.puts(CountBits.bits(10))
IO.puts(CountBits.bits(1234))

File: codewars/java/BitCounting.java
// https://www.codewars.com/kata/526571aae218b8ee490006f4/train/java

import java.util.stream.Collectors;

public class BitCounting {
	public static void main(String[] args) {
		int out = countBits(1234);
		System.out.println(out);
	}

	public static int countBits(int n) {
		return (Integer.toBinaryString(n)).chars().filter(c -> c == '1').mapToObj(Character::toString)
				.collect(Collectors.joining())
				.length();
	}
}
